packages/client/src/modules/pastry/PastryForm.js

```diff
@@ -0,0 +1,159 @@
+import { useMutation } from "@apollo/client";
+import { Button, Grid, TextField, Typography } from "@mui/material";
+import { useFormik } from "formik";
+import React from "react";
+import * as yup from "yup";
+import { SelectCategory } from "../category/SelectCategory";
+import { ADD_PASTRY, UPDATE_PASTRY } from "./mutations";
+
+const fieldStyle = { width: 500 };
+
+const validationSchema = yup.object({
+  name: yup.string().required().label("Name"),
+  origin: yup.string().required().label("Origin"),
+  description: yup.string().required().label("Description"),
+  price: yup.number().required().label("Price").min(0.75),
+  categoryId: yup.string().required().label("Category"),
```

Comment:
The variable name on the server was categoryID, but this is categoryId - this was causing issues for you on the server side with applying updates. In my copy I renamed these all and made one additional change below and the form worked.